test-help-documentation: Skips checking external links test.

sumanthvrao opened the pull request:
This option causes test-help-documentation to verify internal links only.
This prevents documentation from flaking in CI due to links from
external websites having failed in the documentation.
**Before:**
![flake_old](https://user-images.githubusercontent.com/30312566/50374343-a5105600-0612-11e9-9418-99838564f049.png)

**After:**
![flake_new](https://user-images.githubusercontent.com/30312566/50374332-72665d80-0612-11e9-9518-260c28ce2f89.png)
Based on conversation: https://chat.zulip.org/#narrow/stream/43-automated-testing/subject/quora.20404/near/674997

timabbott commented on tools/test-help-documentation:
```diff
@@ -25,14 +29,21 @@ os.makedirs('var/help-documentation', exist_ok=True)
-    ret_api_doc = subprocess.call(('scrapy', 'crawl_with_status', 'api_documentation_crawler'),
-                                  cwd='tools/documentation_crawler')
-    ret_portico_doc = subprocess.call(('scrapy', 'crawl_with_status', 'portico_documentation_crawler'),
+if not options.skip_external_link_check:
```
I think it'd be better to make a varaible, `extra_args`, that we pass into the 3 calls., rather than duplicating so much code.

sumanthvrao commented:
@timabbott I added an extra variable `skip_external` to the three arguments. But i had to change the checking case for `spiders.py`. This works for `test-documentation` as well as `test-help-documentation`.

timabbott commented:
@sumanthvrao I think there's a cleaner way to do the variables; I tweaked and pushed back to this PR.  Can you re-test manually with print statements, and also explain the `crawl` -> `crawl_with_status` change you had made?  I don't understand that piece, and whether it changes the exit status of the scrapy command...

sumanthvrao commented:
I am out of station will test this after couple of days. Thanks for pointing it out. I am still learning to write clean code. I will test it and share the results.

sumanthvrao commented:
@timabbott I manually tested the PR and did some changes. The reason i had initially changed `crawl_with_status` to `crawl` was that the external parameter could only be passed on a direct call to the crawler. `crawl_with_status` indirectly starts another crawler. It also they had the same return value (non zero for failed task).
But to be on the safe side let us use `crawl_with_status` itself. The code which you tweaked did not directly pass the parameter `skip_external` to `kwargs` of  `BaseDocumentationSpider`. The reason which i understood after printing status in different positions, was the way in which the modules were getting called. `test-help-documentation` first called class `HelpDocumentationSpider` which later called `crawl_with_status`. This would then start the crawler with the default parameters ( no external arguments) and hence run `spiders.py` which would receive None value for the variable even though it was passed.
To avoid this, I edited `crawl_with_status` to set a variable skip_external and pass it as a parameter when launching the spider. Here is the [link](https://drive.google.com/file/d/1vjbvdst1q72WkSk2CJhBwWgn80zb3JRn/view?usp=sharing) containing the output of manual testing for `test-help-documentation`.

In PR #10960 also I had used `crawl` instead of `crawl_with_status`. I have added the second commit addressing this. Also changed `test-documentation` where a considerable part of the code was duplicated. Here is the [link](https://drive.google.com/file/d/1TSs_w11JIHXhMG8xrW14AW5Fk34YgLmU/view?usp=sharing) containing the output of manual testing for `test-documentation`

timabbott commented:
Merged, thanks @sumanthvrao!  This is a lot cleaner than the original version, so thanks for doing this cleanup.
